src/TestProblem/Hydro/ClusterMerger/Par_Init_ByFunction_ClusterMerger.cpp

```diff
@@ -488,9 +586,429 @@ void Read_Particles_ClusterMerger( std::string filename, long offset, long num,
+         for (int c=0; c<Merger_Coll_NumHalos; c++) {   
+            num_par_sum[c] = 0; 
+            for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++) {
+               const double *EdgeL = amr->patch[0][lv][PID]->EdgeL;
+               const double *EdgeR = amr->patch[0][lv][PID]->EdgeR;
+               const double patch_pos[3] = { (EdgeL[0]+EdgeR[0])*0.5, (EdgeL[1]+EdgeR[1])*0.5, (EdgeL[2]+EdgeR[2])*0.5 };
+               const double patch_d = sqrt(SQR(EdgeL[0]-EdgeR[0])+SQR(EdgeL[1]-EdgeR[1])+SQR(EdgeL[2]-EdgeR[2]))*0.5;
+               if (SQR(patch_pos[0]-Cen_new_pre[c][0])+SQR(patch_pos[1]-Cen_new_pre[c][1])+SQR(patch_pos[2]-Cen_new_pre[c][2]) <= SQR(20*R_acc+patch_d)){
+                  for (int p=0; p<amr->patch[0][lv][PID]->NPar; p++) {
+                     const long ParID = amr->patch[0][lv][PID]->ParList[p];
+                     const real ParX_tmp = amr->Par->PosX[ParID];
+                     const real ParY_tmp = amr->Par->PosY[ParID];
+                     const real ParZ_tmp = amr->Par->PosZ[ParID];
+                     const real ParM_tmp = amr->Par->Mass[ParID];
+                     const real VelX_tmp = amr->Par->VelX[ParID];
+                     const real VelY_tmp = amr->Par->VelY[ParID];
+                     const real VelZ_tmp = amr->Par->VelZ[ParID];
+                     bool if_cluster = false;
+                     if ( amr->Par->Type[ParID] == real(PTYPE_CLUSTER+c) || amr->Par->Type[ParID] == real(PTYPE_CEN+c) ){
+                        if_cluster = true;
+                     }
+                     if ( if_cluster && SQR(ParX_tmp-Cen_new_pre[c][0])+SQR(ParY_tmp-Cen_new_pre[c][1])+SQR(ParZ_tmp-Cen_new_pre[c][2]) <= SQR(10*R_acc) ){
+//                      Record the mass, position and velocity of this particle
+                        ParX[c][num_par[c]] = ParX_tmp;
+                        ParY[c][num_par[c]] = ParY_tmp;
+                        ParZ[c][num_par[c]] = ParZ_tmp;
+                        ParM[c][num_par[c]] = ParM_tmp;
+                        VelX[c][num_par[c]] = VelX_tmp;
+                        VelY[c][num_par[c]] = VelY_tmp;
+                        VelZ[c][num_par[c]] = VelZ_tmp;
+                        num_par[c] += 1; 
+                     }
+                     if (num_par[c] >= N) {
+                         N = num_par[c] + 1;  // Increase the new maximum size if needed
+                         ParX[c] = (double*)realloc(ParX[c], N * sizeof(double));
+                         ParY[c] = (double*)realloc(ParY[c], N * sizeof(double));
+                         ParZ[c] = (double*)realloc(ParZ[c], N * sizeof(double));
+                         ParM[c] = (double*)realloc(ParM[c], N * sizeof(double));
+                         VelX[c] = (double*)realloc(VelX[c], N * sizeof(double));
+                         VelY[c] = (double*)realloc(VelY[c], N * sizeof(double));
+                         VelZ[c] = (double*)realloc(VelZ[c], N * sizeof(double));
```

Comment:
[Bug] These arrays are allocated by `new` in c++, which could be incompatible with `realloc` in c. You need to stick with either the c++ or c approach for memory manipulation.